### ISSUE TITLE
Electron webRequest APIs vs Chrome webRequest APIs

### ISSUE THREAD

**Issue by sachinjain024** [electron/electron#17602]
Hey guys,

I am new to electron framework. I just want to understand how electron APIs are related to chrome extension/apps APIs. 

**For example**, When I checked [electron webRequest API](https://electronjs.org/docs/api/web-request), I could see the API signature and surface area matches quite closely with [chrome webRequest APIs](https://developer.chrome.com/extensions/webRequest). 

As I know, Chromium (read as Google as though Chromium is open source but decisions/investments are majorly driven by Google) team is doing some major changes to webRequest APIs as discussed here - 
1. https://groups.google.com/a/chromium.org/forum/#!topic/chromium-extensions/veJy9uAwS00%5B1-25%5D
2. https://docs.google.com/document/d/1nPu6Wy4LWR66EFLeYInl3NzzhHzc-qnk4w4PX-0XMw8/preview#heading=h.3wdnvzqkvonh
3. https://bugs.chromium.org/p/chromium/issues/detail?id=896897&desc=2

In short, they are planning to take away blocking capabilities from webRequest APIs and introducing new APIs - declarativeNetAPIs.

Once rolled out, will such a change also affect Electron APIs or not? In general, any changes to Chrome APIs does affect Electron Apps or not?

**Comment by robatwilliams** [comment]
This is in the news today. It's stated that enterprise customers will be able to use the webRequest API in Chromium, so I'd expect it to still be there and therefore probably usable by Electron without huge divergence.

**Comment by nornagon** [comment]
Electron's `webRequest` API does not share code with Chrome extensions. It does depend on some particular internals of Chrome's network stack, but there are no planned changes to that in the foreseeable future that would prevent Electron's `webRequest` API from working.
